netdata-installer.sh

```diff
@@ -1022,6 +1027,174 @@ install_go() {
 
 install_go
 
+function get_kernel_version() {
+  r="$(uname -r | cut -f 1 -d '-')"
+
+  read -r -a p <<< "$(echo "${r}" | tr '.' ' ')"
+
+  printf "%03d%03d%03d" "${p[0]}" "${p[1]}" "${p[2]}"
+}
+
+detect_libc() {
+  libc=
+  if ldd --version 2>&1 | grep -q -i glibc; then
+    echo >&2 " Detected GLIBC"
+    libc="glibc"
+  elif ldd --version 2>&1 | grep -q -i 'gnu libc'; then
+    echo >&2 " Detected GLIBC"
+    libc="glibc"
+  elif ldd --version 2>&1 | grep -q -i musl; then
+    echo >&2 " Detected musl"
+    libc="musl"
+  else
+    echo >&2 " ERROR: Cannot detect a supported libc on your system!"
+    return 1
+  fi
+  echo "${libc}"
+  return 0
+}
+
+get_compatible_kernel_for_ebpf() {
+  kver="${1}"
+
+  # XXX: Logic taken from Slack discussion in #ebpf
+  # everything that has a version <= 4.14 can use the code built to 4.14
+  # Continue the logic, everything that has a version <= 4.19 and >= 4.15 can use the code built to 4.19
+  # Finally,  everybody that is using 5.X can use what is compiled with 5.4
+
+  kpkg=
+
+  if [ "${kver}" -ge 005000000 ]; then
+    echo >&2 " Using eBPF Kernel Package built against Linux 5.4"
+    kpkg="5_4"
+  elif [ "${kver}" -ge 004015000 ] && [ "${kver}" -le 004020017 ]; then
+    echo >&2 " Using eBPF Kernel Package built against Linux 4.19"
+    kpkg="4_19"
+  elif [ "${kver}" -le 004014999 ]; then
+    echo >&2 " Using eBPF Kernel Package built against Linux 4.14"
+    kpkg="4_14"
+  else
+    echo >&2 " ERROR: Cannot detect a supported kernel on your system!"
+    return 1
+  fi
+
+  echo "${kpkg}"
+  return 0
+}
+
+should_install_ebpf() {
+  if [ "${NETDATA_ENABLE_EBPF:=0}" -ne 1 ]; then
+    run_failed "ebpf not enabled. --enable-ebpf to enable"
```

Comment:
should this be failed?

It looks like an error during install which is not

```
>  --- eBPF Kernel Collector (opt-in) ---
 FAILED  ebpf not enabled. --enable-ebpf to enable
```

Reply:
Is there a better function to use here? I couldn't find one; so I'll probably just change this to `echo >&2 "..."`

Reply:
https://github.com/netdata/netdata/compare/master...prologic:ebpf_installer_tweaks?expand=1